saleor/userprofile/forms.py

```diff
@@ -15,13 +16,18 @@ def get_address_form(data, country_code, initial=None, instance=None, **kwargs):
         country_code = country_form.cleaned_data['country']
         preview = country_form.cleaned_data['preview']
 
+    if data:
```

Comment:
What I am struggling right now is to transfer user input to the form, so it's handled properly (by 'properly' I mean to glue `phoneprefix` and `phonenumber` and doing a reverse - after providing address data, user can go back and select another address - proper fields should be repopulated. It could be done with a lot of `ifs` (checking if instance exists, if it has `phone` attribute, etc), but it seems to be very non-elegant and messy solution.

Reply:
You'll find it rather challenging to handle this as two separate fields. I think it would be much easier if it was a single field (as it's going to be stored as such) with a [MultiWidget](https://docs.djangoproject.com/en/2.0/ref/forms/widgets/#django.forms.MultiWidget).

In fact there is already one [provided by the library you're using](https://github.com/stefanfoulis/django-phonenumber-field/blob/master/phonenumber_field/widgets.py#L39).

Reply:
Such a great tip, thank you, solved many issues I had!